storage-s3-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java

```diff
@@ -103,9 +138,12 @@ public void write(
         if (overwrite) {
             writeActions(fs, path, actions);
             return;
+        } else if (fs.exists(path)) {
```

Comment:
> This validation better be as close as possible to the part where it's added to the external log-store.

Why? We have changed our TTL such that it is an hour (but @zsxwing wants it changed to a day) after the write. We assume that our file system cannot provide mutual exclusion on writes .... but it can detect after several minutes (and therefore, also for an hour or a day) if a file already exists in the FileSystem.

Reply:
The new ttl does solve it.
But for the sake of users who upgraded delta but didn’t notice they need to update the ttl definition, better minimize the risky time

Reply:
That's a good point! Yep dropping `commitTime` is safer. We also need to write a migration guide for the next release to highlight the recommended TTL change.